### PR TITLE
TAStudio cleanup

### PR DESCRIPTION
**Pull request by SuuperW**
I cleaned up some movie loading code in TAStudio.
I'd like other users to test for any issues these changes might cause.

### CODE REVIEW

**Comment by adelikat** [comment]
need to readdress this after the zwinderstatemanager branch merge

Did you test not saving default.tasproj?  It was doing that for a reason, otherwise how do you do a core reboot when they pick new project?

**Comment by SuuperW** [comment]
You're right, not saving the file is causing problems. I could have sworn I tested that, though...

**Comment by adelikat** [comment]
I committed the same change before too, swore I tested too, and got so many bugs on the tracker :D

**Comment by SuuperW** [comment]
From new commit's code comment:
		We want to create a new file for the movie, unless the original movie was the default movie.
		Cluttering up the user's movie folder with default 1.tasproj, default 2.tasproj, etc. isn't nice.
		We also want to promt the user for a file name if we haven't saved a file here, so we use the default name.
		Now, playing a movie loads the movie from the file; which would mean playing the new movie actually loads the default movie.
		However if the movie's file doesn't exist, then it simply skips that loading process!
		But I find it strange that it doesn't fail to play the movie without a file, so maybe this isn't an ideal solution?

I do not know if this skipping of the load process for non-existent files is meant to actually work. If it is, then a better solution is probably to make that explicit by including this code at the top of [Bk2Movie.IO.cs -> Load](https://github.com/TASVideos/BizHawk/blob/a4e3dbb8855868062f93b49c3415d177a9cd924a/src/BizHawk.Client.Common/movie/bk2/Bk2Movie.IO.cs#L30):
`if (!File.Exists(Filename)) return true;`
The we could give the movie a filename of null after conversion, which looks like it would (along with a change to auto-save code to check for null filename) would also have the desired effect of prompting the user for a file name upon saving.

File: src/BizHawk.Client.EmuHawk/tools/TAStudio/TAStudio.cs [hunk 1]
```diff
@@ -529,39 +529,32 @@ private void ConvertCurrentMovieToTasproj()
...
-				GoToFrame(0);
-			}
-			else if (gotoFrame > 0)
-			{
-				GoToFrame(gotoFrame);
-			}
-			else
-			{
-				GoToFrame(CurrentTasMovie.TasSession.CurrentFrame);
-			}
```
Comment (RetroEdit) [inline src/BizHawk.Client.EmuHawk/tools/TAStudio/TAStudio.cs:564]: Would removing this prevent the tasproj from loading and returning the current frame cursor to the old position? (see also: #2181)

(Haven't tested it.)

Comment (SuuperW) [reply src/BizHawk.Client.EmuHawk/tools/TAStudio/TAStudio.cs:564]: Probably. I couldn't think of any possible use case where that behavior would be desired, and saw no code or comments that appeared to utilize or explain it.

Why would you ever want to re-load the current movie, and want it to automatically seek to the previously loaded frame?

(I certainly can add this back in, it just seemed like a messy bit of code with no explanation whatever so I cleaned it.)

Comment (RetroEdit) [reply src/BizHawk.Client.EmuHawk/tools/TAStudio/TAStudio.cs:564]: I meant when you opened the movie for the first time: the cursor is supposed to remember what frame the movie was on the previous time the movie was saved.

I do think the way it was previously implemented looked a bit messy and could be improved.

Comment (SuuperW) [reply src/BizHawk.Client.EmuHawk/tools/TAStudio/TAStudio.cs:564]: Oh, that makes sense. I will revert that change.

**Comment by YoshiRulz** [comment]
New state manager was merged and bugs squished, is any of this still relevant?

**Comment by SuuperW** [comment]
No, I don't think any of this is still needed.

**Comment by RetroEdit** [comment]
Then this PR should be closed. I'm not keen on releasing code that was written right before a major refactor of a related component without

**Pull request closed without merging**